catalyst-toolbox/src/rewards/proposers/types.rs

```diff
@@ -0,0 +1,130 @@
+use jormungandr_lib::crypto::hash::Hash;
+use serde::{Deserialize, Serialize};
+use std::{path::PathBuf, str::FromStr};
+use structopt::StructOpt;
+
+use color_eyre::{eyre::eyre, Report};
+
+macro_rules! bool_enum {
+    ($enum_name:ident, $true_case:ident, $false_case:ident) => {
+        #[derive(Debug, Serialize, Deserialize)]
+        #[serde(rename_all = "UPPERCASE")]
+        pub enum $enum_name {
+            $true_case,
+            $false_case,
+        }
+
+        impl From<bool> for $enum_name {
+            fn from(b: bool) -> Self {
+                match b {
+                    true => $enum_name::$true_case,
+                    false => $enum_name::$false_case,
+                }
+            }
+        }
+    };
+}
+
+bool_enum!(YesNo, Yes, No);
+bool_enum!(FundedStatus, Funded, NotFunded);
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum NotFundedReason {
+    #[serde(rename = "Not Funded - Over Budget")]
+    OverBudget,
+    #[serde(rename = "Not Funded - Approval Threshold")]
+    ApprovalThreshold,
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
```

Comment:
clap would be nice in the future.

Reply:
Yep, we're stuck on a version before the derive API was added IIRC. But clap 4 just got released :/